src/components/timebank/timebank-content.tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  Box,
+  List,
+  ListItem,
+  Select,
+  MenuItem,
+  FormControlLabel,
+  Checkbox,
+  ListItemText,
+  Typography,
+  Card,
+  CircularProgress
+} from "@mui/material";
+import { formatTimePeriod, getHoursAndMinutes } from "../../utils/time-utils";
+import type { KeycloakProfile } from "keycloak-js";
+import { DailyEntry, Timespan } from "../../generated/client";
+import TimebankPieChart from "../charts/timebank-piechart";
+import TimebankOverviewChart from "../charts/timebank-overviewchart";
+import { DatePicker } from "@mui/x-date-pickers";
+import { DateTime } from "luxon";
+import strings from "../../localization/strings";
+import TimebankMultiBarChart from "../charts/timebank-multibarchart";
+import DateRangePicker from "./timebank-daterange-picker";
+import { useState } from "react";
+import { theme } from "../../theme";
+import { useAtomValue } from "jotai";
+import { personTotalTimeAtom, personDailyEntryAtom, dailyEntriesAtom } from "../../atoms/person";
+
+interface Props {
+  userProfile: KeycloakProfile | undefined;
+  timespanSelector: Timespan;
+  getPersonTotalTime: (timespan?: Timespan) => void;
+  handleDailyEntryChange: (selectedDate: DateTime | null) => void;
+  loading: boolean;
+}
+
+/**
+ * Component that contains the entirety of Timebank content, such as charts
+ */
+const TimebankContent = (props: Props) => {
+  const { timespanSelector, getPersonTotalTime, handleDailyEntryChange, loading } = props;
+
+  const [selectedEntries, setSelectedEntries] = useState<DailyEntry[]>([]);
+  const [byRange, setByRange] = useState({
+    dailyEntries: false
+  });
+  const personTotalTime = useAtomValue(personTotalTimeAtom);
+  const personDailyEntry = useAtomValue(personDailyEntryAtom);
+  const dailyEntries = useAtomValue(dailyEntriesAtom);
+  const today = DateTime.fromJSDate(
+    dailyEntries.filter((item) => item.date.getMonth() <= new Date().getMonth())[0].date
+  );
+
+  /**
+   * Allows only logged dates or with expected hours to be selected in the date time picker.
+   *
+   * @param date DateTime object passed from the date picker
+   * @returns boolean value which controls the disabled dates.
+   */
+  const disableNullEntries = (date: DateTime): boolean => {
+    const loggedDates = dailyEntries.filter(
+      (item) =>
+        item.logged &&
+        item.expected &&
+        DateTime.fromJSDate(item.date).toISODate() === date.toISODate()
+    )[0];
+
+    return loggedDates
+      ? !(DateTime.fromJSDate(loggedDates.date).toISODate() === date.toISODate())
+      : true;
+  };
+
+  /**
+   * Renders overview chart and list item elements containing total time summaries
+   */
+  const renderOverViewChart = () => {
+    if (loading) return <CircularProgress sx={{ margin: "auto", mt: "5%", mb: "5%" }} />;
```

Comment:
As a general rule when returning JSX/ components in a conditional like this, return from the condition body (so add curly brackets)